### DISCUSSION TITLE
How to use DepthPickingPass when logarithmicDepthBuffer is set？

### DISCUSSION THREAD

**Discussion by 53356454** [pmndrs/postprocessing#626] [Q&A]
When the renderer sets logarithmicDepthBuffer to true, the depth value obtained using DepthPickingPass is incorrect. How can I obtain the correct depth value? (Performance can be disregarded) Thanks

**Comment by vanruesc** [comment]
The `DepthPickingPass` currently doesn't support log depth out of the box. The return value is the depth value from the depth texture. It should be possible to decode that value with something like this:

```js
const { near, far } = camera;
const d = Math.pow(2.0, logDepth * Math.log2(far + 1.0)) - 1.0;
const a = far / (far - near);
const b = far * near / (near - far);
const depth = a + b / d;
```

I haven't tested this though.
